File: 57-insert-interval/insert-interval.js
/**
 * @param {number[][]} intervals
 * @param {number[]} newInterval
 * @return {number[][]}
 */
var insert = function(intervals, newInterval) {
    let merged = false, [x1, y1] = newInterval, result = []

    for (let [x, y] of intervals) {
        if (merged) {
            result.push([x, y]);
        } else if (y1 < x) {
            result.push([x1, y1], [x, y])
            merged = true;
        } else if (x1 > y) {
            result.push([x, y])
        } else {
            x1 = Math.min(x1, x);
            y1 = Math.max(y1, y)
        }
    }

    if (!merged) result.push([x1, y1])
    return result;
};

File: 57-insert-interval/insert-interval.swift
class Solution {
    func insert(_ intervals: [[Int]], _ newInterval: [Int]) -> [[Int]] {
        var merged: Bool = false, result: [[Int]] = []
        var cx: Int = newInterval[0], cy: Int = newInterval[1]
        for arr: [Int] in intervals {
            guard !merged else {
                result.append(arr)
                continue
            }
            let x: Int = arr[0], y: Int = arr[1] 
            if x >= cx && x <= cy || y >= cx && y <= cy || cx >= x && cx <= y || cy >= x && cy <= y {
                print("merged")
                cx = min(x, cx); cy = max(y, cy)
            } else if cy < x {
                result.append([cx, cy])
                result.append([x, y])
                merged = true
            } else {
                result.append([x, y])
            }
        }
        if !merged { result.append([cx, cy]) }
        return result
    }
}